apl-utils/src/main/java/com/apollocurrency/aplwallet/apl/util/task/DefaultTaskDispatcher.java

```diff
@@ -0,0 +1,412 @@
+/*
+ * Copyright © 2018-2019 Apollo Foundation
+ */
+package com.apollocurrency.aplwallet.apl.util.task;
+
+import lombok.Getter;
+import lombok.extern.slf4j.Slf4j;
+import org.jboss.weld.util.collections.ListMultimap;
+import org.jboss.weld.util.collections.Multimap;
+
+import java.util.*;
+import java.util.concurrent.*;
+import java.util.concurrent.atomic.AtomicInteger;
+
+@Slf4j
+public class DefaultTaskDispatcher implements TaskDispatcher {
+
+    public static final int DEFAULT_THREAD_POOL_SIZE = 10;
+    public static final String APL_BG_WORKERS = "apl-bg-workers";
+    public static final String APL_POOL_NAME = "apl-dispatcher";
+
+    private static final AtomicInteger groupNumber = new AtomicInteger(1);
+
+    @Getter
+    protected final Map<String,String> initParameters;
+
+    protected ExecutorServiceFactory executorServiceFactory;
+    protected TaskExecutorService backgroundThread;
+    @Getter
+    protected String serviceName;
+
+    protected volatile boolean started = false;
+
+    private ExecutorService onStartExecutor;
+    private Multimap<TaskOrder, Task> tasks = new ListMultimap<>();
```

Comment:
Any reason to use Multimap instead of HashMap?

Reply:
If we need more than one type of tasks like INIT, BEFORE or AFTER we have to use a list of tasks. Replaced to HashMap.